pkg/reconciler/namespace/controller.go

```diff
@@ -53,15 +58,34 @@ func NewController(
 	ctx context.Context,
 	cmw configmap.Watcher,
 ) *controller.Impl {
-
+	logger := logging.FromContext(ctx)
 	namespaceInformer := namespace.Get(ctx)
 	serviceAccountInformer := serviceaccount.Get(ctx)
 	roleBindingInformer := rolebinding.Get(ctx)
 	brokerInformer := broker.Get(ctx)
 	configMapPropagationInformer := configmappropagation.Get(ctx)
 
+	recorder := controller.GetEventRecorder(ctx)
+	if recorder == nil {
```

Comment:
this stuff inside the if is the one we will be duplicating in different reconcilers. I think I saw it already in the source CRD or the duck one... anyway we can put it in a common place?

Reply:
na, it is just a few lines and when we can generate k8s types it will be deleted.

Reply:
Yeah, mtnamespace needs this too.  We also have the same problem in nscert in Serving.  It'd be good to get this, even if it means YOLO hand rolling ReconcileKind for K8s namespaces for the time being.

Reply:
I can do that.